### ISSUE TITLE
Google Adwords API Integration

### ISSUE THREAD

**Issue by Nithiyaomprakash** [googleads/googleads-php-lib#597]
Hi,

My team has been trying to connect to Google Adwords through an API to get the reporting information out on to an integrated dashboard.

We have now activated a Basic API on the interface which should suffice our need.

While connecting to the API, we get the attached message. Can you direct us to someone who can help us in this regard?

![image_2019_05_21T12_58_15_067Z](https://user-images.githubusercontent.com/52067971/59910438-82c78b80-942f-11e9-8e1d-58fc4da1b66a.png)

**Comment by fiboknacky** [comment]
Have you installed this library using `composer` as suggested in [the README](https://github.com/googleads/googleads-php-lib#downloading-this-library)?

**Comment by fiboknacky** [comment]
And it seems you're trying to send a request to v201802, which is already sunset. Even if you can make the client library works, the request to the server will fail.
I suggest start using [this library](https://github.com/googleads/google-ads-php), which is for the Google Ads API, instead.

**Comment by Nithiyaomprakash** [comment]
Thanks for your suggestion, will check and get back to you!

**Comment by Nithiyaomprakash** [comment]
Hi, below the error

![image](https://user-images.githubusercontent.com/52067971/59916496-f0c67f80-943c-11e9-884f-3f307b191ae4.png)

**Comment by fiboknacky** [comment]
Could you please answer my questions above first, so I know how to help you?
Thank you.

**Comment by Nithiyaomprakash** [comment]
Yes we did!

**Comment by fiboknacky** [comment]
I don't see you answer to [this question](https://github.com/googleads/googleads-php-lib/issues/597#issuecomment-504352444) yet. 
Could you please confirm?

**Comment by Nithiyaomprakash** [comment]
Can you repeat your question.

**Comment by fiboknacky** [comment]
"Have you installed this library using composer as suggested in the README?"

**Comment by Nithiyaomprakash** [comment]
Yes we installed as per suggestion in README.
ᐧ

**Comment by Nithiyaomprakash** [comment]
Already PHP library and composer (version 1.8.5) successfully installed.

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
Thank you. Did you install using `composer require` or did you download the compressed tarball?
Could you please elaborate where you created your file and how your file is organized under the directory together with the vendor/ directory?

**Comment by Nithiyaomprakash** [comment]
The composed version has installed and used the vendor directory.

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ
ᐧ

**Comment by fiboknacky** [comment]
May I suggest answering my questions one by one and in detail? In that way, I could help fixing your issues faster and we can reduce back-and-forth. :-)

For my above questions,

> - Did you install using `composer require` or did you download the compressed tarball?

Which one did you do?

> Could you please elaborate where you created your file and how your file is organized under the directory together with the vendor/ directory?

Can you include much more details? I know that you're using the vendor directory, but I cannot see how your files are organized, so I cannot help further.

**Comment by Nithiyaomprakash** [comment]
- Did you install using composer require or did you download the
   compressed tarball?

We installed composer require.

Could you please elaborate where you created your file and how your file is
organized under the directory together with the vendor/ directory?


[image: image.png]





[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
I can't see your image. Could you check please?

**Comment by Nithiyaomprakash** [comment]
[image: image.png]

Please confirm now you can able to see.

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
No, it looks like you answer using an email?
If you log in GitHub, you can see that the image is not there.
Please attach an image when you logged in on GitHub. Thank you.

**Comment by Nithiyaomprakash** [comment]
Hi,

Here the file.

![Api](https://user-images.githubusercontent.com/52067971/60016421-c4606c80-96a3-11e9-953f-29c3919c4f5b.PNG)
![Api](https://user-images.githubusercontent.com/52067971/60016431-cd513e00-96a3-11e9-85ee-22a896a7d829.PNG)

**Comment by fiboknacky** [comment]
That is the structure of this library. But where is your file?
And this doesn't look like you've installed via `composer require`.
It's more like you downloaded this library directly and then do `composer install` in the library directory.

Can you confirm please?

**Comment by Nithiyaomprakash** [comment]
we are checking on that

**Comment by Nithiyaomprakash** [comment]
![image](https://user-images.githubusercontent.com/52067971/60069527-cde3e600-9730-11e9-837b-9f21516e751f.png)

**Comment by fiboknacky** [comment]
I'd repeat my question again. Can you answer them _one by one_ please?

*  Where is your file that you use to run the script and find this error? (I'm not asking about composer.phar)
* Could you confirm you use `composer require` to install this library? 
* Or did you download this library via zip file, for example?

Thank you.

**Comment by Nithiyaomprakash** [comment]
I will check with my developer once again

**Comment by fiboknacky** [comment]
Feel free to loop your developer in this thread so we can resolve this issue more quickly. Thanks.

**Comment by Nithiyaomprakash** [comment]
He is not comfortable on this, He is next to me only each and everything I am sharing with him.

**Comment by Nithiyaomprakash** [comment]
is this possible to have a call with you

**Comment by fiboknacky** [comment]
I'm sorry. We're providing support via GitHub issue trackers and our Google Ads API forum only.
If you provide all the information I asked for, I'm quite sure we can resolve this issue fast enough. :)

**Comment by Nithiyaomprakash** [comment]
1. Where is your file that you use to run the script and find this error?

http://wisoftsolutions.net/googleapinets/examples/BasicOperations/GetCampaigns.php


Fatal error: Class 'Google\Ads\GoogleAds\Examples\Utils\ArgumentParser' not found in /home1/sajinair/public_html/wisoftsolutions.net/googleapinets/examples/BasicOperations/GetCampaigns.php on line 38

2. Could you confirm you use composer require to install this library? - 
We used composer require to install this library.

3. Or did you download this library via zip file, for example?

  zip file download - https://github.com/googleads/googleads-php-lib

**Comment by fiboknacky** [comment]
You used a "composer require" OR "download" the library?
It's an OR. You cannot do both.
If you do both, then you're doing wrong.

Can you confirm please?

**Comment by Nithiyaomprakash** [comment]
my developer is doing both

**Comment by Nithiyaomprakash** [comment]
can you guide him by step by step? If possible please the steps.

**Comment by fiboknacky** [comment]
Then, he was doing this wrongly. :)
Please find the step-by-step in the README file. In particular, you need to follow either of the three ways described in [this section](https://github.com/googleads/googleads-php-lib#downloading-this-library).

**Comment by Nithiyaomprakash** [comment]
Using composer require

$ composer require googleads/googleads-php-lib
![image](https://user-images.githubusercontent.com/52067971/60078693-165bcd80-974a-11e9-943d-0f5d49e846d7.png)

**Comment by Nithiyaomprakash** [comment]
we want to know the next step, please guide us

**Comment by fiboknacky** [comment]
The next step is all in the README file. Could you please follow [them](https://github.com/googleads/googleads-php-lib#using-composer-require) and let me know specifically where you get stuck?

**Comment by fiboknacky** [comment]
Provided that you have all OAuth2 credentials and developer token ,you can now start writing your code.

**Comment by Nithiyaomprakash** [comment]
Hi

My team got stuck here, please help them to move next step.

![image](https://user-images.githubusercontent.com/52067971/60164005-64d89d00-981a-11e9-9764-c7027647b963.png)

**Comment by Nithiyaomprakash** [comment]
they dont know what to do next

**Comment by Nithiyaomprakash** [comment]
![image](https://user-images.githubusercontent.com/52067971/60167852-8db06080-9821-11e9-9282-6bf2c20c63e9.png)

**Comment by fiboknacky** [comment]
Can you explain in details what your team have done?
With only screenshots, it's hard to follow what else have you done and how you followed our instructions.

**Comment by Nithiyaomprakash** [comment]
https://github.com/googleads/google-ads-php

git clone https://github.com/googleads/google-ads-php.git

cd google-ads-php

Run composer install at the command prompt. This will install all dependencies needed for using the library and running examples.

Set up your OAuth2 credentials.

If you have the adsapi_php.ini file you used for the AdWords API, copy and name it as google_ads_php.ini. Simply change the section name from [ADWORDS] to [GOOGLE_ADS].

php examples/BasicOperations/GetCampaigns.php --customerId <YOUR_CUSTOMER_ID>

Fatal error: Class 'Google\Ads\GoogleAds\Examples\Utils\ArgumentParser' not found in /home1/sajinair/public_html/wisoftsolutions.net/adsapi/google-ads-php/examples/BasicOperations/GetCampaigns.php on line 43

**Comment by fiboknacky** [comment]
What you described belong to [another library](https://github.com/googleads/google-ads-php), not this one.
Could you ensure you've followed this library's README?

**Comment by Nithiyaomprakash** [comment]
Hi, We followed the instruction and used the same library.

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
No, you probably misunderstand.

What you described above belong to the another library named, `google-ads-php` (for the Google Ads API).
This one is `googleads-php-lib` (the AdWords API).

Could you confirm again what library you wish to use?

**Comment by Nithiyaomprakash** [comment]
For google leads

We had clone the following from https://github.com/googleads/google-ads-php

git clone https://github.com/googleads/google-ads-php.git

(Google Ads API Client Library for PHP)

As mentioned there,

we did the following

=>First installed composer.

=>Cloned project in the directory via:

git clone https://github.com/googleads/google-ads-php.git

=>Changed into the google-ads-php directory.

cd google-ads-php

=>Executed 'composer install' at the command prompt.

=>Set up  OAuth2 credentials.

=>Run the GetCampaigns example to test if credentials are valid.

=>Getting the error "Fatal error: Class
'Google\Ads\GoogleAds\Examples\Utils\ArgumentParser' not found in
/home1/sajinair/public_html/
wisoftsolutions.net/googleapinets/examples/BasicOperations/GetCampaigns.php
on line 38"


Note: Please let me know where we are missing!

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
Before going forward, could you please answer the following question?

**What you described above belong to the another library named, google-ads-php (for the Google Ads API).
This one is googleads-php-lib (the AdWords API).**

**Comment by Nithiyaomprakash** [comment]
What you mean by Google ads and Google leads. We want to pull below data
from Google ads interface.

Impression
Clicks
Cost
Conversion
CTR
CPC
Cost/conversion

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by Nithiyaomprakash** [comment]
You suggest me which one we have to consider.

[image: nithiya]
[image: wisoft logo] <http://www.wisoftsolutions.com/>
<https://www.facebook.com/Wisoft-Solutions-557566661072063/>
<https://twitter.com/Wisoftsolutions>
<https://www.linkedin.com/company/1351121?trk=prof-exp-company-name>
*Nithiya Omprakash*
*Digital Advertising Manager*
P: +971 4 3210710 <+97143210710>   F: +971 4380 4244 <+97143804244>
*Wisoft Solutions*
P.O Box 237159, Office M-38, Curve Building, Sheikh Zayed Rd, Dubai UAE
www.wisoftsolutions.com
<https://www.wisoftsolutions.com/hubspot-marketing-automation/>
<https://www.wisoftsolutions.com/leadsquared-marketing-automation/>
<https://www.wisoftsolutions.com/ppc-advertising/>

ᐧ

**Comment by fiboknacky** [comment]
[Google Ads API](https://developers.google.com/google-ads/api/docs/start) is the new version of the AdWords API, so I'd suggest you start using it instead.
Because this GitHub repo is for the AdWords API, please file a new bug on [the google-ads-php library](https://github.com/googleads/google-ads-php) instead if you still face with any installation problems.
